src/codegen/object.rs

```diff
@@ -178,10 +181,24 @@ fn generate_builder(w: &mut dyn Write, env: &Env, analysis: &analysis::object::I
                     "Vec<GString>" => "Vec<String>",
                     typ => typ,
                 };
-                let (param_type, conversion) = match type_string {
-                    "String" => ("&str", ".to_string()"),
-                    typ => (typ, ""),
+                let mut param_type = rust_type_full(
+                    env,
+                    property.typ,
+                    library::Nullable(false),
+                    property.set_in_ref_mode,
+                    library::ParameterScope::None,
+                    library::Concurrency::None,
+                )
+                .unwrap(); //Must be Ok if rust_type is Ok
+                let (param_type_override, conversion) = match &param_type[..] {
+                    "&str" => ("", ".to_string()"),
+                    "&[&str]" => ("Vec<String>", ""),
+                    typ if typ.starts_with('&') => ("", ".clone()"),
+                    _ => ("", ""),
                 };
+                if !param_type_override.is_empty() {
```

Comment:
I'd rather use `None` instead of the empty string and do:
```
if let Some(typ) = param_type_override {
```

Reply:
You right.
Done